lib/plug/csrf_protection.ex

```diff
@@ -304,19 +313,29 @@ defmodule Plug.CSRFProtection do
         mode == :clear_session ->
           conn |> configure_session(ignore: true) |> clear_session()
 
-        mode == :exception ->
-          raise InvalidCSRFTokenError
-
-        true ->
+        mode not in @valid_options ->
           raise ArgumentError,
                 "option :with should be one of :exception or :clear_session, got #{inspect(mode)}"
+
+        invalid_session?(conn, csrf_token) ->
```

Comment:
This is also raising for when there is a session but the session itself is invalid - which you can see in the change tests. Also, I am worried that changing the error can be a backwards incompatible change. What if we tackle this problem by improving the error message instead? The message can say:

```
Please make sure that:

  * The session cookie is being sent and session is loaded
  * The CSRF token is sent either via request parameter or HTTP header
```

WDYT?